### PR TITLE
Query: Add support for owned reference navigation

### PR DESCRIPTION
**Pull request by smitpatel**
Part of #15285

### CODE REVIEW

File: src/EFCore.Relational/Query/Pipeline/EntityProjectionExpression.cs [hunk 1]
```diff
@@ -107,5 +111,31 @@ public ColumnExpression BindProperty(IProperty property)
...
+                    $"Called EntityProjectionExpression.AddNavigationBinding() with incorrect INavigation. " +
+                    $"EntityType:{EntityType.DisplayName()}, Property:{navigation.Name}");
+            }
+
+            _navigationExpressionsCache[navigation] = entityShaper;
+        }
+
+        public EntityShaperExpression BindNavigation(INavigation navigation)
+        {
+            if (!EntityType.GetTypesInHierarchy().Contains(navigation.DeclaringEntityType))
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Query/Pipeline/EntityProjectionExpression.cs]:
```suggestion
            if (!EntityType.IsAssignableFrom(navigation.DeclaringEntityType) 
                && !navigation.DeclaringEntityType.IsAssignableFrom(EntityType))
```

File: src/EFCore.Relational/Query/Pipeline/EntityProjectionExpression.cs [hunk 2]
```diff
@@ -107,5 +111,31 @@ public ColumnExpression BindProperty(IProperty property)
 
             return expression;
         }
+
+        public void AddNavigationBinding(INavigation navigation, EntityShaperExpression entityShaper)
+        {
+            if (!EntityType.GetTypesInHierarchy().Contains(navigation.DeclaringEntityType))
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Query/Pipeline/EntityProjectionExpression.cs]:
```suggestion
            if (!EntityType.IsAssignableFrom(navigation.DeclaringEntityType) 
                && !navigation.DeclaringEntityType.IsAssignableFrom(EntityType))
```

File: test/EFCore.Specification.Tests/Query/OwnedQueryTestBase.cs [hunk 3]
```diff
@@ -574,6 +580,8 @@ protected override void OnModelCreating(ModelBuilder modelBuilder, DbContext con
                                             });
                                     });
                             });
+
+                        eb.Ignore(p => p.Orders);
```
Comment (AndriySvyryd) [inline test/EFCore.Specification.Tests/Query/OwnedQueryTestBase.cs]: Don't ignore base navigations on derived types

File: src/EFCore.Relational/Query/Pipeline/EntityProjectionExpression.cs (hunk 1 above)
Comment (smitpatel) [reply src/EFCore.Relational/Query/Pipeline/EntityProjectionExpression.cs]: Does not work. It also need to check if the entityType is towards base.

Comment (AndriySvyryd) [reply src/EFCore.Relational/Query/Pipeline/EntityProjectionExpression.cs]: Can you elaborate? The proposed code should behave exactly the same as the old.

Comment (smitpatel) [reply src/EFCore.Relational/Query/Pipeline/EntityProjectionExpression.cs]: I missed the second line of change :(

**Comment by smitpatel** [comment]
Will work on collections after your PR is merged.

**Pull request merged**